protocol/interop-monitoring.md

```diff
@@ -0,0 +1,132 @@
+- Double checking that every executing message is valid
+
+### ExecutingMessages and Access Lists
+
+The [access list](https://github.com/ethereum-optimism/design-docs/blob/9e919c5b173fe8fc89949b012f6f70a0bc3247f6/protocol/interop-access-list.md)
+design guarantees the fact that all executing messages can be validated without the need to execute the transaction. Any calls to the `CrossL2Inbox`
+that do not include the statically declared executing message in the access list will revert rather than needing to be dropped. This prevents
+a denial of service attack where the MEV searcher can simply produce an invalid `ExecutingMessage` after their MEV attempt fails.
+
+Given that the decided upon approach depends strictly on the current EVM resource pricing via storage slot cost introspection, we should have
+monitoring to alert us if someone is able to trick the `CrossL2Inbox` into producing an `ExecutingMessage` when the access list entry is
+not declared. We think this is impossible, but given this is such a critical security property, it is important to monitor.
+
+### Double Checking Message Validity
+
+#### Unsafe Blocks
+
+We want to utilize the cloud architecture in [this doc](https://github.com/ethereum-optimism/design-docs/pull/218) to ensure that
+no invalid `ExecutingMessage`s are ever included in a block. No matter what tradeoffs we make, it is impossible to guarantee there
+will not be a contingent reorg because an unsafe head reorg can happen after all cross chain transaction validity checks passed.
+
+We want to be able to detect when an invalid `ExecutingMessage` is included in an unsafe block and trigger an altert to the
+oncall engineering team. Additionally, we should consider pausing the batcher automatically if an invalid `ExecutingMessage`
+has been detected in an unsafe block and triggering an unsafe head reorg. It is preferable to not waste blobs and trigger
+an unsafe head reorg by batch submitting the invalid block. The problem with this is that it is indistinguishable from
+a malicious sequencer triggering a reorg to extract MEV. Either way an unsafe head reorg is going to happen, its just whether
+or not its due to the data being posted and then resulting in a replacement deposits only block or if its manually done
+by the sequencer offchain.
+
+We may also want to consider a way to alert partners in the interop set ahead of time that an unsafe head reorg is coming
+if an invalid `ExecutingMessage` is observed in an unsafe block. If they turn off their cross chain message ingress fast enough,
+it could be possible that they can prevent a contingent reorg. The liveness of the chain can continue with no issues until the
+remote chain goes through its unsafe head reorg, then it can open up its cross chain message ingress again.
+
+#### Safe/Finalized Blocks
+
+If an invalid `ExecutingMessage` ends up in a safe block, that means that a bug becomes consensus (without multiclient).
```

Comment:
I'm going to be taking an editorial pass on this doc, but just wanted to point out this is not the case -- invalid messages may appear in Locally Safe blocks, but if they are in *Cross Safe* blocks there is a bugged consensus.